Subject: Remove static from may_create() and may_delete().

This patch allows LSM to use may_create() and may_delete().

Signed-off-by: Kentaro Takeda <takedakn@nttdata.co.jp>
Signed-off-by: Tetsuo Handa <penguin-kernel@I-love.SAKURA.ne.jp>
Signed-off-by: Toshiharu Harada <haradats@nttdata.co.jp>
---
 fs/namei.c         |    5 ++---
 include/linux/fs.h |    2 ++
 2 files changed, 4 insertions(+), 3 deletions(-)

--- vfs.orig/fs/namei.c
+++ vfs/fs/namei.c
@@ -1475,8 +1475,7 @@ static inline int check_sticky(struct in
  * 10. We don't allow removal of NFS sillyrenamed files; it's handled by
  *     nfs_async_unlink().
  */
-static int may_delete(struct dentry *dir_dentry, struct dentry *victim,
-		      int isdir)
+int may_delete(struct dentry *dir_dentry, struct dentry *victim, int isdir)
 {
 	struct inode *dir = dir_dentry->d_inode;
 	int error;
@@ -1517,7 +1516,7 @@ static int may_delete(struct dentry *dir
  *  3. We should have write and exec permissions on dir
  *  4. We can't do it if dir is immutable (done in permission())
  */
-static inline int may_create(struct dentry *dir_dentry, struct dentry *child)
+int may_create(struct dentry *dir_dentry, struct dentry *child)
 {
 	if (child->d_inode)
 		return -EEXIST;
--- vfs.orig/include/linux/fs.h
+++ vfs/include/linux/fs.h
@@ -1791,6 +1791,8 @@ extern void free_write_pipe(struct file 
 extern struct file *do_filp_open(int dfd, const char *pathname,
 		int open_flag, int mode);
 extern int may_open(struct nameidata *, int, int);
+extern int may_create(struct dentry *, struct dentry *);
+extern int may_delete(struct dentry *, struct dentry *, int);
 
 extern int kernel_read(struct file *, unsigned long, char *, unsigned long);
 extern struct file * open_exec(const char *);
